### PR TITLE
[Windows] Add app startup integration test

### PR DESCRIPTION
**Pull request by loic-sharma**
Add a basic integration test for the Windows's app startup. In a subsequent change (https://github.com/flutter/flutter/pull/109816), this test will be updated to verify the app's window isn't visible until after the first frame has been drawn.

Commits:
1. https://github.com/flutter/flutter/commit/49f54e0728a2ff6c844c960e4c25bb0e83a4583d - Run `flutter create`. No need to review this!
2. https://github.com/flutter/flutter/pull/110114/commits/2ee770e6107ede467c75136dde6b30b2410086f8 - Prepare new project for changes
3. https://github.com/flutter/flutter/pull/110114/commits/49f54e0728a2ff6c844c960e4c25bb0e83a4583d - Add app startup integration test

Part of https://github.com/flutter/flutter/issues/41980

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/integration_tests/windows_startup_test/test_driver/main_test.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+void main() {
+  test('Windows app starts and draws frame', () async {
+    final FlutterDriver driver = await FlutterDriver.connect(printCommunication: true);
+    final String result = await driver.requestData(null);
+
+    expect(result, equals('success'));
+
+    await driver.close();
+  }, timeout: Timeout.none);
+}
```
Comment (yaakovschectman) [inline dev/integration_tests/windows_startup_test/test_driver/main_test.dart:17]: I think this will need a newline to pass the tests.

**Review by yaakovschectman — APPROVED** [review]
LGTM code-wise other than failing tests, possibly all due to the one newline.

File: dev/integration_tests/windows_startup_test/test_driver/main_test.dart (hunk 1 above)
Comment (loic-sharma) [reply dev/integration_tests/windows_startup_test/test_driver/main_test.dart]: Nice catch, fixed!

File: dev/devicelab/bin/tasks/windows_startup_test.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_devicelab/framework/devices.dart';
+import 'package:flutter_devicelab/framework/framework.dart';
+import 'package:flutter_devicelab/tasks/integration_tests.dart';
+
+Future<void> main() async {
+  deviceOperatingSystem = DeviceOperatingSystem.windows;
+  await task(createWindowsStartupDriverTest(deviceIdOverride: 'windows'));
```
Comment (loic-sharma) [inline dev/devicelab/bin/tasks/windows_startup_test.dart:11]: Please let me know if this `deviceIdOverride` isn't necessary

File: .ci.yaml [hunk 3]
```diff
@@ -4181,3 +4181,17 @@ targets:
       tags: >
         ["devicelab", "android", "windows"]
       task_name: windows_chrome_dev_mode
+
+  - name: Windows windows_startup_test
```
Comment (a-wallen) [inline .ci.yaml:4185]: @yaakovschectman previously named a test to contain `win_desktop`. You two should coordinate for consistency.

**Review by a-wallen — APPROVED** [review]

File: .ci.yaml (hunk 3 above)
Comment (loic-sharma) [reply .ci.yaml:4185]: It looks like Yaakov's test is consistent with example apps' start up tests. However, other tests use a different pattern. For example, the test named `Windows platform_channel_sample_test_windows` has a task named `platform_channel_sample_test_windows`. I'll match that

File: dev/integration_tests/windows_startup_test/windows/flutter/generated_plugin_registrant.cc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+//
```
Comment (cbracken) [inline dev/integration_tests/windows_startup_test/windows/flutter/generated_plugin_registrant.cc:1]: These are ephemeral files that we shouldn't check in.

File: dev/integration_tests/windows_startup_test/windows/flutter/generated_plugins.cmake [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#
```
Comment (cbracken) [inline dev/integration_tests/windows_startup_test/windows/flutter/generated_plugins.cmake:1]: This one gets checked in though!

File: dev/integration_tests/windows_startup_test/lib/main.dart [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  // Create a completer to send the result back to the integration test.
+  final Completer<String> completer = Completer<String>();
+  enableFlutterDriverExtension(handler: (String? message) => completer.future);
+
+  try {
+    const MethodChannel methodChannel =
+        MethodChannel('tests.flutter.dev/windows_startup_test');
+
+    final bool? visible = await methodChannel.invokeMethod('isWindowVisible');
+    if (visible == null || visible == false) {
```
Comment (cbracken) [inline dev/integration_tests/windows_startup_test/lib/main.dart:42]: Maybe just `if (visible != true) {`?

File: dev/integration_tests/windows_startup_test/lib/main.dart [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      throw 'Window should be visible at startup';
+    }
+
+    ui.PlatformDispatcher.instance.onBeginFrame = (Duration duration) async {
+      final bool? visible = await methodChannel.invokeMethod('isWindowVisible');
+      if (visible == null || visible == false) {
```
Comment (cbracken) [inline dev/integration_tests/windows_startup_test/lib/main.dart:48]: Similar comment.

File: dev/integration_tests/windows_startup_test/lib/main.dart (hunk 6 above)
Comment (cbracken) [inline dev/integration_tests/windows_startup_test/lib/main.dart:42]: Shouldn't the window _not_ be visible at startup (until beginFirstFrame)?

Comment (loic-sharma) [reply dev/integration_tests/windows_startup_test/lib/main.dart:42]: I would love to do `if (visible != true)` but there's a [Dart lint that warns if you implicitly handle the null](https://dart.dev/tools/linter-rules#use_if_null_to_convert_nulls_to_bools); they recommend something like `if (!(visible ?? false))` or `if ((visible ?? false) == false)` which I find too terse. This felt like the most reasonable compromise. Does that seem reasonable? Or should I ignore that lint?

File: dev/integration_tests/windows_startup_test/windows/flutter/generated_plugins.cmake (hunk 5 above)
Comment (loic-sharma) [reply dev/integration_tests/windows_startup_test/windows/flutter/generated_plugins.cmake:1]: Removed the other file but kept this one!

File: dev/integration_tests/windows_startup_test/lib/main.dart (hunk 6 above)
Comment (cbracken) [reply dev/integration_tests/windows_startup_test/lib/main.dart:42]: Ah yeah -- I had a vague recollection of something in our style guide I vehemently disliked about these sorts of checks. This is better than either of those. No worries then :)

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

File: dev/integration_tests/windows_startup_test/lib/main.dart (hunk 6 above)
Comment (loic-sharma) [reply dev/integration_tests/windows_startup_test/lib/main.dart:44]: (Sorry forgot to reply to this comment!)

Hiding the app's window until after the first frame was moved into its own pull request: https://github.com/flutter/flutter/pull/109816

**Pull request merged**